### PR TITLE
PMK-2619, rework header row for infrastructure tab

### PR DESCRIPTION
**Pull request by jruddell**
image

image

image

### CODE REVIEW

File: src/app/plugins/kubernetes/components/common/ListPageHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        <UsageWidget title="Memory" stats={totals.memory} units="GiB" />
+        <UsageWidget title="Storage" stats={totals.disk} units="GiB" />
+      </div>
+    </>
+  )
+}
+
+export default ListPageHeader
+
+const useStyles = makeStyles<Theme>((theme) => ({
```
Comment (brennancheung) [inline src/app/plugins/kubernetes/components/common/ListPageHeader.tsx]: Should this be moved to top for consistency?

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ClustersListPageHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const curUsagePaths = {
+  computeCurrPath: 'usage.compute.current',
+  memoryCurrPath: 'usage.memory.current',
+  diskCurrPath: 'usage.disk.current',
+}
+
+export const ClusterListHeader = () => {
+  const handleGetUsage = useCallback((clusters: ICluster[]) => {
+    const items = clusters.filter((cluster) => (cluster.nodes || []).length > 0)
+    return getUsageTotals(items, curUsagePaths, maxUsagePaths)
```
Comment (brennancheung) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/ClustersListPageHeader.tsx]: `getUsageTotals` seems like it fits a more general purpose pattern.  Might not need to hard-code the keys if they are already in the paths objects.

File: src/app/plugins/kubernetes/util/calcUsageTotals.js [hunk 3]
```diff
@@ -29,4 +30,16 @@ const calcUsageTotals = curry((items, currentPathStr, maxPathStr) => {
...
-export default calcUsageTotals
+export default calcUsageTotalByPath
+
+export const getUsageTotals = moize(
+  (
+    items,
+    { computeCurrPath, memoryCurrPath, diskCurrPath },
+    { computeMaxPath, memoryMaxPath, diskMaxPath },
+  ) => ({
+    compute: calcUsageTotalByPath(items, computeCurrPath, computeMaxPath),
```
Comment (brennancheung) [inline src/app/plugins/kubernetes/util/calcUsageTotals.js]: keys might not need to be hard-coded if path objects can already contain the key names.

File: src/app/plugins/kubernetes/components/common/ListPageHeader.tsx (hunk 1 above)
Comment (jruddell) [reply src/app/plugins/kubernetes/components/common/ListPageHeader.tsx]: I hate having to scroll past all the styles to get to a component. Maybe we should move them all below the component definition?

File: src/app/plugins/kubernetes/util/calcUsageTotals.js (hunk 3 above)
Comment (jruddell) [reply src/app/plugins/kubernetes/util/calcUsageTotals.js]: Hmm, the paths are different based on the type of entity being passed. `nodes.combined.usage`. `clusters.usage`. What do you mean by hard-coded keys? The keys are passed in so they can be variable.

**Pull request merged**